sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/_delete_polling.py

```diff
@@ -44,13 +43,13 @@ def run(self):
 
     def finished(self):
         # type: () -> bool
-        if self._sd_disabled:
+        if not self._deleted_resource.recovery_id:
             return True
         return self._status == "deleted"
 
     def resource(self):
-        # type: () -> DeletedKey
-        return self._deleted_key
+        # type: () -> Union[DeletedKey, DeletedSecret]
```

Comment:
This will need to be `Any` until someone takes the time to express it as a [generic protocol](https://mypy.readthedocs.io/en/latest/generics.html#generic-protocols).